esper/assets/js/views/video.jsx

```diff
@@ -111,13 +112,33 @@ class VideoView extends React.Component {
     );
   }
 
+  _handlePaginationSelect(selectedPage){
+	  this.setState({
+		  activePage: selectedPage
+	  });
+  }
+	  
+
   _renderLabeler() {
     let video = this.props.store;
+	const stride = Math.ceil(video.fps)/3;
     let all_boxes = [];
+	// TODO: doing something generic here (passing a list of items)
+	// is challenging becaues of the way bounding boxes are drawn
+	// I will revisit this when this is set in stone
+	const framesPerPage = 200;
+	const totalPages = video.num_frames/(stride*framesPerPage);
+	const activePage = this.state.activePage;
+	const firstFrame = activePage*framesPerPage*stride;
+	const lastFrame = Math.min(firstFrame+(framesPerPage*stride), video.num_frames);
+
 
     return (
       <div className='video-labeler'>
-        {_.range(0, video.num_frames, 24).map((n, ni) => {
+	  <div>
+      <Pagination prev next first last ellipsis boundaryLinks maxButtons={10} activePage={activePage+1} items={totalPages} onSelect={(x) => {this._handlePaginationSelect(x-1)}}/>
```

Comment:
Same point I mentioned earlier: make sure you don't create an inline closure for performance reasons.